src/server/pfs/server/driver.go

```diff
@@ -1461,35 +1454,83 @@ func (d *driver) createBranch(ctx context.Context, branch *pfs.Branch, commit *p
 				// possible that branch exists but has no head commit. This is fine, but
 				// branchInfo.Head must also be nil
 				if _, ok := err.(pfsserver.ErrNoHead); !ok {
-					return err
+					return fmt.Errorf("inspectCommit: %v", err)
 				}
+				commit = nil
+			} else {
+				commit = commitInfo.Commit
 			}
 		}
 		branchInfo := &pfs.BranchInfo{
 			Branch: branch,
 			Name:   branch.Name,
 		}
-		if commitInfo != nil {
-			branchInfo.Head = commitInfo.Commit
+		// Get the previous version of this branch if there is one, we do this
```

Comment:
I'm still working on understanding "pre-existing subvenance" (and this code in general), but updating a branch shouldn't change its subvenance, right? This function takes a `provenance` argument (which, btw, it might make sense to call `directProvenance` now), so those branches' subvenance may change, but this branch's subvenance (and downstream branches' subvenance) should stay the same.